### DISCUSSION TITLE
Team mode: use of colors to locate team mates

### DISCUSSION THREAD

**Discussion by ymollard** [streetcomplete/StreetComplete#5862] [Ideas]
Hey, I tried the team mode and have a couple of questions/suggestions:
- should not team mode deactivate by itself for a certain team member after some inactivity? Isn't there a risk to miss quests if a team member forgets to disable this mode?
- what are the colors of each team member used for?
- could colors be used to locate your team mates on the map? That questions privacy for sure, but as soon as you are warned that your location will been shared during mapping and team mode deactivates when you are idle, that looks ok to me. And very handy in order not to loose your team during mapping
- are/should some rewards dedicated to team mapping to encourage more collective sessions?

**Comment by matkoniecz** [comment]
> what are the colors of each team member used for?

coordination between people (who selects which colour)

> could colors be used to locate your team mates on the map? 

not really as no data is exchanged between phones

**Comment by mnalis** [comment]
To answer some more of your questions:

> - [...] Isn't there a risk to miss quests if a team member forgets to disable this mode?

I wouldn't say so. There is a notification in the upper right corner of the main screen all the time while team mode is active, and there are other way mapper can forget to change (changing list of quests, remaining in non-default preset, etc) that would result in less quests being shown. 

And nothing horrible will happen even if they forget to exit the team mode - they'd simply get a little more exercise for same amount of solved quests (which might not be altogether a bad thing anyway :smiley: )

> - what are the colors of each team member used for?

They determine what quests you get, in order so each mapper gets different set of quests. For example, if you select two-people team, first color would get all quests pertaining to objects with even-numbered id, and second color would get all quests pertaining to the objects having odd-numbered id. 

In other words, if by mistake two mappers happened to choose the same color, they would get exactly the same quests, i.e. they would be solving exactly the same questions, which would waste the effort (as answers to already solved quests are simply discarded). The fix would be simple once they noticed they are solving the same quests -- one of them would exit the team mode and enter it again, choosing same number of team players, but picking a still unused color/letter.

> - are/should some rewards dedicated to team mapping to encourage more collective sessions?

Generally, see ["SC FAQ: I like gamification factor of StreetComplete, and have an idea how to increase it even more"](https://wiki.openstreetmap.org/wiki/StreetComplete/FAQ#I_like_gamification_factor_of_StreetComplete,_and_have_an_idea_how_to_increase_it_even_more), which explains why it might be problematic, especially for things like team competitions, leaderboards etc. 

Sociologically, there might perhaps be an acceptable/unproblematic extra achievement like _**"Team player"**_, which gains levels the more quests you solve while in team mode, but I think there might be technical issues implementing that (as uploaded changesets do not contain information whether they were done in team mode or not, such information is then likely not available for statistics purposes upon which achievements are built upon either).

**Reply by ymollard** [reply]
- Nothing horrible happens if I do not deactivate my team mode. But if I never open the app again, my quests will remain assigned to me and never be answered by anyone?
- My question about colors was more focused about the color itself, because you could discriminate the users/quests using their phone ID for instance, instead of requesting them to pick a color. But I understand that it could be useful to make sure that the teammates talk together and cooperate.
- Yes I imagined a "Team player" badge, nothing that pushes competition, since I read that FAQ question :p

**Reply by matkoniecz** [reply]
> But if I never open the app again, my quests will remain assigned to me and never be answered by anyone?

There is no reservation, team mode only limits what is shown on your phone. 

Team mode info does not leave your phone.

You need to select your colour manually instead of getting one assigned - as it is quest assignment method with such manual communication
